p2p/dht.go

```diff
@@ -157,3 +161,34 @@ func (q BlobstreamDHT) GetValsetConfirm(ctx context.Context, key string) (types.
 	}
 	return confirm, nil
 }
+
+// PutLatestValset encodes a valset then puts its value to the DHT.
+// The key will be returned by the `GetValsetKey` method.
+// If the valset is not the latest, it will fail.
+// Returns an error if it fails.
+func (q BlobstreamDHT) PutLatestValset(ctx context.Context, v types2.Valset) error {
+	encodedData, err := types.MarshalValset(v)
+	if err != nil {
+		return err
+	}
+	err = q.PutValue(ctx, GetLatestValsetKey(), encodedData)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetLatestValset looks for the latest valset in the DHT.
+// The key will be returned by the `GetValsetKey` method.
+// Returns an error if it fails.
+func (q BlobstreamDHT) GetLatestValset(ctx context.Context) (types2.Valset, error) {
+	encoded, err := q.GetValue(ctx, GetLatestValsetKey()) // this is a blocking call, we should probably use timeout and channel
+	if err != nil {
+		return types2.Valset{}, err
+	}
+	valset, err := types.UnmarshalValset(encoded)
+	if err != nil {
+		return types2.Valset{}, err
+	}
+	return valset, nil
+}
```

Comment:
iirc, you mentioned that the the main concern with this approach is that we aren't verifying the valdiator set that is gossipped. Would that be possible though (in a futuer PR ofc)? We should be able to query the current validator set, and from that we should be able to measure the diff in power between the one we recieve that this one afaiu.

does that make sense or am I missing something?

Reply:
yes, we could, but I didn't want to add that check because it will be checked implicitly when sending the update to Blobstream. So, no need to do it explicitly here

Reply:
do you mind pointing out that check really quickly for my own understanding? thanks!

Reply:
It will be done here:

https://github.com/celestiaorg/blobstream-contracts/blob/72b11fbc5ebd10c11e518bfd8b3f3a806d343b59/src/Blobstream.sol#L269-L276

in the contract

Reply:
but is it possible to gossip and invalid value? also, isn't the issue for folks deploying a new contract? would that check still be hit?

Reply:
yes, an invalid value can be gossiped, but will be rejected by the contract when relaying events.

For the deployment part, I didn't implement the ability to use the P2P valset to deploy because it's super risky and there is no way to trust it.